### PR TITLE
hcm config: Split FilterChainTest into separate file

### PR DESCRIPTION
**Pull request by auni53**
Commit Message:
* Split test functions into config_test_base.h
* Move FilterChainTest tests into config_filter_chain_test.cc
* Other tests from config_test.cc can be moved in the future to improve readability
Risk Level: Low

As part of #14470, I'll need to add filter dependency validation logic for http filterchains. This is a cleanup CL to make that easier.

### CODE REVIEW

File: test/extensions/filters/network/http_connection_manager/config_test_base.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#include "test/extensions/filters/network/http_connection_manager/config.pb.h"
+#include "test/extensions/filters/network/http_connection_manager/config.pb.validate.h"
+#include "test/mocks/config/mocks.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/server/factory_context.h"
+#include "test/test_common/printers.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/test_runtime.h"
+#include "test/test_common/utility.h"
```
Comment (htuch) [inline test/extensions/filters/network/http_connection_manager/config_test_base.h:27]: There are a lot of headers here, might be nice to shrink them based on IWYU principles.

File: test/extensions/filters/network/http_connection_manager/config_filter_chain_test.cc [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::Eq;
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace HttpConnectionManager {
```
Comment (htuch) [inline test/extensions/filters/network/http_connection_manager/config_filter_chain_test.cc:19]: Nit: tests should live in the anonymous namespace in here.

**Review by htuch — COMMENTED** [review]
Looks great, thanks for the cleanup! Some minor comments.

File: test/extensions/filters/network/http_connection_manager/config_filter_chain_test.cc (hunk 2 above)
Comment (auni53) [reply test/extensions/filters/network/http_connection_manager/config_filter_chain_test.cc:19]: Done.

File: test/extensions/filters/network/http_connection_manager/config_test_base.h (hunk 1 above)
Comment (auni53) [reply test/extensions/filters/network/http_connection_manager/config_test_base.h]: Did my best, but it's pretty time-consuming and inaccurate doing this manually what with all the unseen nested dependencies.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Review by htuch — APPROVED** [review]

**Pull request merged**